clouds/azure/utils.py

```diff
@@ -0,0 +1,54 @@
+from typing import List
+from simple_logger.logger import get_logger
+from azure.core.exceptions import HttpResponseError
+from azure.mgmt.subscription import SubscriptionClient
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.redhatopenshift import AzureRedHatOpenShiftClient
+
+LOGGER = get_logger(name=__name__)
+
+
+def get_aro_supported_versions(aro_client: AzureRedHatOpenShiftClient, region: str) -> List[str]:
+    supported_versions = [aro_version.version for aro_version in aro_client.open_shift_versions.list(location=region)]
+    LOGGER.info(f"ARO supported versions: {supported_versions}")
+    return supported_versions
+
+
+def get_azure_supported_regions(subscription_client: SubscriptionClient, subscription_id: str) -> List[str]:
+    supported_regions = [
+        region.name for region in subscription_client.subscriptions.list_locations(subscription_id=subscription_id)
+    ]
+    LOGGER.info(f"ARO supported regions: {supported_regions}")
+    return supported_regions
+
+
+def azure_resources_cleanup(resource_client: ResourceManagementClient) -> None:
+    """
+    Azure cloud resources are associated with resource groups.
+    Running this cleanup function will destroy all existing leftover resources
+    that are associated with the given resource client's subscription.
+    Please note that this action is irreversible.
+    Args:
+        resource_client: Azure resources client (see https://learn.microsoft.com/en-us/python/api/azure-mgmt-resource/azure.mgmt.resource.resources.resourcemanagementclient?view=azure-python)
+                            for more info.
+    Returns:
+        None
+    """
+    failed_delete_resource_groups = []
```

Comment:
every const that you define should have type.

`failed_delete_resource_groups: List = [str]`

Reply:
done